### PR TITLE
chore(application-payments): Refactor payments to a separate lib

### PR DESCRIPTION
**Pull request by obmagnusson**
## What

Move payments application logic to a separate lib. 
Payments using  V2 of client [merged PR from Origo](https://github.com/island-is/island.is/pull/9353 )

## Why

To pave the way for more reusability in applications from the template api.

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/api/domains/payment/src/lib/api-domains-payment.module.ts [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
+  imports: [
+    ChargeFjsV2ClientModule,
+    ConfigModule.forRoot({
+      isGlobal: true,
+      load: [XRoadConfig, ChargeFjsV2ClientConfig],
+    }),
+  ],
+  providers: [PaymentResolver],
+})
+export class ApiDomainsPaymentModule {}
```
Comment (eirikurn) [inline libs/api/domains/payment/src/lib/api-domains-payment.module.ts:20]: Feel free to add yourself as codeowner of this api module.

File: libs/api/domains/payment/src/lib/api-domains-payment.module.ts [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
-    }
-  }
-}
+@Module({
+  imports: [
+    ChargeFjsV2ClientModule,
+    ConfigModule.forRoot({
+      isGlobal: true,
+      load: [XRoadConfig, ChargeFjsV2ClientConfig],
+    }),
```
Comment (eirikurn) [inline libs/api/domains/payment/src/lib/api-domains-payment.module.ts:16]: This should only be imported in the app module.

**Review by eirikurn — APPROVED** [review]
Approved core files with minor comments.

**Review by egill15 — APPROVED** [review]
LGTM

File: libs/application/template-api-modules/src/lib/modules/templates/marriage-conditions-submission/marriage-conditions-submission.service.ts [hunk 3]
```diff
@@ -79,9 +79,12 @@ export class MarriageConditionsSubmissionService extends BaseTemplateApiService
     application: { id },
     auth,
   }: TemplateApiModuleActionProps) {
+    const SYSLUMADUR_NATIONAL_ID = '6509142520'
```
Comment (kksteini) [inline libs/application/template-api-modules/src/lib/modules/templates/marriage-conditions-submission/marriage-conditions-submission.service.ts:82]: Just a thought: Seeing this pop up in a couple of places which makes me think this should be stored in a constants file somewhere.

File: libs/application/template-api-modules/src/lib/modules/templates/driving-license-duplicate/driving-license-duplicate.service.ts [hunk 4]
```diff
@@ -25,12 +25,15 @@ export class DrivingLicenseDuplicateService {
     application: { id, answers },
     auth,
   }: TemplateApiModuleActionProps) {
+    const SYSLUMADUR_NATIONAL_ID = '6509142520'
```
Comment (kksteini) [inline libs/application/template-api-modules/src/lib/modules/templates/driving-license-duplicate/driving-license-duplicate.service.ts:28]: Like here

File: libs/application/template-api-modules/src/lib/modules/templates/driving-license-submission/driving-license-submission.service.ts [hunk 5]
```diff
@@ -36,6 +36,8 @@ export class DrivingLicenseSubmissionService extends BaseTemplateApiService {
     application: { id, answers },
     auth,
   }: TemplateApiModuleActionProps) {
+    const SYSLUMADUR_NATIONAL_ID = '6509142520'
```
Comment (kksteini) [inline libs/application/template-api-modules/src/lib/modules/templates/driving-license-submission/driving-license-submission.service.ts:39]: and here

File: libs/application/template-api-modules/src/lib/modules/templates/example-payment-actions/examplePaymentActions.service.ts [hunk 6]
```diff
@@ -18,10 +18,13 @@ export class ExamplePaymentActionsService extends BaseTemplateApiService {
     application: { id, answers },
     auth,
   }: TemplateApiModuleActionProps) {
+    // Performing organization ID
+    const SYSLUMADUR_NATIONAL_ID = '6509142520'
```
Comment (kksteini) [inline libs/application/template-api-modules/src/lib/modules/templates/example-payment-actions/examplePaymentActions.service.ts:22]: and so on

**Review by kksteini — APPROVED** [review]
Reviewed CO files. LGTM

**Review by brynjarorng — APPROVED** [review]
LGTM (Devops)

**Pull request merged**